src/logger.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+/**
+ * @module logger
+ */
+const log = require("loglevel");
+
+// This is to demonstrate, that you can use any namespace you want.
+// Namespaces allow you to turn on/off the logging for specific parts of the
+// application.
+// An idea would be to control this via an environment variable (on Node.js).
+// See https://www.npmjs.com/package/debug to see how this could be implemented
+// Part of #332 is introducing a logging library in the first place.
+const DEFAULT_NAME_SPACE = "matrix";
+const logger = log.getLogger(DEFAULT_NAME_SPACE);
+log.setDefaultLevel(log.levels.WARN);
```

Comment:
This is creating and exporting the 'matrix' logger but setting the default log level on the root logger, so I don't think the `setDefaultLevel` actually does anything here.

Also we probably want to maintain the current behaviour by default (for now, at least) but setting the default level to WARN would turn them off (`log.log` being a synonym for `log.debug`). I'd probably suggest using `log.info` for the current messages and setting the default level to DEBUG?

Reply:
From the docs:
> * A log.setDefaultLevel(level) method.
> This sets the current log level only if one has not been persisted and can’t be loaded. This is useful when initializing scripts; if a developer or user has previously called setLevel(), this won’t alter their settings.

I'll adjust it nevertheless to reduce confusion.

Reply:
Yep, but I think they're still independent between loggers as far as I can tell? ie. `log.setDefaultLevel()` sets a level if one hasn't been set with `log.setLevel()`, and affects `log.log()`, but will not affect `logger.setLevel()` or `logger.log()`? Anyway, if they're all using `logger` this is fine.